pypeit/core/flexure.py

```diff
@@ -432,135 +628,115 @@ def spec_flexure_slit(slits, slitord, slit_bpm, sky_file, method="boxcar", speco
+        else:
+            spec_fwhm = None
 
-        # Down the center of the slit (spec_method = slitcen)
         if slit_cen:
-            sky_wave = slit_specs[islit].wavelength.value
-            sky_flux = slit_specs[islit].flux.value
-
-            # Calculate the shift
-            fdict = spec_flex_shift(slit_specs[islit], sky_spectrum, arx_fwhm_pix,
-                                    mxshft=mxshft, excess_shft=excess_shft, spec_fwhm=spec_fwhm)
-            # Failed?
-            if fdict is not None:
-                # Update dict
-                for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
-                    flex_dict[key].append(fdict[key])
-                # Interpolate
-                sky_wave_new = flexure_interp(fdict['shift'], sky_wave)
-                flex_dict['sky_spec'].append(xspectrum1d.XSpectrum1D.from_tuple((sky_wave_new, sky_flux)))
-                flex_dict['method'].append("slitcen")
-            else:
-                msgs.warn(f'Flexure corrections cannot be performed for slit # {slits.spat_id[islit]}')
-
-        # Along the boxcar extraction (spec_method = boxcar)
+            # global flexure
+            flex_list = spec_flex_shift_global(slit_specs, islit, sky_spectrum, arx_fwhm_pix, empty_flex_dict,
+                                               return_later_slits, flex_list, keys_to_update,
+                                               spec_fwhm=spec_fwhm, mxshft=mxshft, excess_shft=excess_shft)
         else:
-            i_slitord = slitord[islit]
-            indx = specobjs.slitorder_indices(i_slitord)
-            this_specobjs = specobjs[indx]
-            if len(this_specobjs) == 0:
-                msgs.info('No object extracted in this slit.')
+            # local flexure
+            flex_list = spec_flex_shift_local(slits, slitord, specobjs, islit, sky_spectrum, arx_fwhm_pix,
+                                              empty_flex_dict, return_later_slits, flex_list, keys_to_update,
+                                              spec_fwhm=spec_fwhm, mxshft=mxshft, excess_shft=excess_shft)
+
+    # Check if we need to go back to some failed slits
+    if len(return_later_slits) > 0:
+        msgs.warn(f'Flexure shift calculation failed for {len(return_later_slits)} slits')
+        # take the median value to deal with the cases when there are more than one shift per slit (e.g., local flexure)
+        saved_shifts = np.array([np.percentile(flex['shift'], 50, interpolation='nearest')
+                                 if len(flex['shift']) > 0 else None for flex in flex_list])
+        if np.all(saved_shifts == None):
+            # If all the elements in saved_shifts are None means that there are no saved shifts available
+            msgs.warn(f'No previously saved flexure shift estimates available. '
+                      f'Flexure corrections cannot be performed.')
```

Comment:
If this line is to be printed on a separate line, you could use ` ...available.' + msgs.newline() + f'Flexure corrections...`